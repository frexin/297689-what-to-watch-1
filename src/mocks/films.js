const movies = [
  {
    id: 1, name: `Bohemian Rhapsody`, picture: `bohemian-rhapsody.jpg`, genre: `Crime`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 2, name: `Aviator`, picture: `aviator.jpg`, genre: `Dramas`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`

  },
  {
    id: 3, name: `What We Do in the Shadows`, picture: `what-we-do-in-the-shadows.jpg`, genre: `Crime`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 4, name: `Snatch`, picture: `snatch.jpg`, genre: `Horror`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 5, name: `Revenant`, picture: `revenant.jpg`, genre: `Romance`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

export {movies};
