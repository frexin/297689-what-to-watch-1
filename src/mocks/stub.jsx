import React from "react";

const StubComponent = () => {
  return (<div/>);
};

export default StubComponent;
