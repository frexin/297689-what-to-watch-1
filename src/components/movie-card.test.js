import React from 'react';
import renderer from 'react-test-renderer';

import {movies} from "../mocks/films";
import MovieCard from "./movie-card";

it(`MovieCard component loads correctly`, () => {
  const mv = movies[0];
  const tree = renderer.create(<MovieCard preview={mv.preview} id={mv.id} name={mv.name} picture={mv.picture} onMovieSelect={()=>{}}/>);

  expect(tree.toJSON()).toMatchSnapshot();
});
