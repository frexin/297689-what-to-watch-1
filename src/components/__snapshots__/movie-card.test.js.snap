// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component loads correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <button
    className="small-movie-card__play-btn"
    onClick={[Function]}
    type="button"
  >
    Play
  </button>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Bohemian Rhapsody"
      height="175"
      src="img/bohemian-rhapsody.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
    >
      Bohemian Rhapsody
    </a>
  </h3>
</article>
`;
